index.js

```diff
@@ -27,7 +39,70 @@ app.use(express.json());
 // IDEAS METHODS
 
 app.get('/ideas', (req, res) => {
-    res.status(200).json(getIdeas());
+    console.log(req.query);
+    try {
+        if (req.query.sortBy === 'totalDonationSum') {
+            res.status(200).json(sortedByDonationSumIdeas());
+            return;
+        }
+        if (req.query.sortBy === 'status' && req.query.status === 'accepted') {
+            res.status(200).json(approvedIdeasList());
+            return;
+        }
+        if (req.query.sortBy === 'status' && req.query.status === 'pending') {
+            res.status(200).json(pendingIdeasList());
+            return;
+        }
+        if (req.query.sortBy === 'status' && req.query.status === 'rejected') {
+            res.status(200).json(rejectedIdeasList());
+            return;
+        }
+        res.status(200).json(getIdeas());
```

Comment:
Try to simplify and call `res.status(200)` only once.

Reply:
Updated and tested